File: repair_encoding.js

const fs = require('fs');
const path = require('path');

try {
    const srcPath = path.resolve(__dirname, 'src');
    if (!fs.existsSync(srcPath)) {
        fs.writeFileSync('repair_error.txt', `SRC path does not exist: ${srcPath}`);
        process.exit(1);
    }

    const patterns = [
        { s: /\u00C3\u00A9/g, r: 'é' },
        { s: /\u00C3\u00A1/g, r: 'á' },
        { s: /\u00C3\u00B3/g, r: 'ó' },
        { s: /\u00C3\u00AA/g, r: 'ê' },
        { s: /\u00C3\u00A7/g, r: 'ç' },
        { s: /\u00C3\u00A3/g, r: 'ã' },
        { s: /\u00C3\u00A0/g, r: 'à' },
        { s: /\u00C3\u00B5/g, r: 'õ' },
        { s: /\u00C3\u00BA/g, r: 'ú' },
        { s: /\u00C3\u00AD/g, r: 'í' },
        { s: /\u00C3\u0089/g, r: 'É' },
        { s: /\u00C3\u0081/g, r: 'Á' },
        { s: /\u00C3\u0093/g, r: 'Ó' },
        { s: /\u00C3\u0087/g, r: 'Ç' },
        { s: /\u00C3\u0095/g, r: 'Õ' },
        { s: /\u00C3\u009A/g, r: 'Ú' },
        { s: /\u00C3\u008D/g, r: 'Í' },
        { s: /\u00C0--/g, r: 'x' },
        { s: /Observaces/g, r: 'Observações' },
        { s: /movimentaces/g, r: 'movimentações' },
        { s: /manifestá/g, r: 'manifesto' },
        { s: /inventrio/g, r: 'inventário' },
        { s: /prescriptionNão/g, r: 'prescriptionNumber' }
    ];

    let log = 'Repair Started\n';
    function walk(dir) {
        const files = fs.readdirSync(dir);
        for (const file of files) {
            const fullPath = path.join(dir, file);
            if (fs.statSync(fullPath).isDirectory()) {
                if (file !== 'node_modules' && file !== '.git' && file !== 'dist') {
                    walk(fullPath);
                }
            } else if (/\.(tsx|ts|css)$/.test(file)) {
                let content = fs.readFileSync(fullPath, 'utf8');
                let modified = false;
                patterns.forEach(p => {
                    if (p.s.test(content)) {
                        content = content.replace(p.s, p.r);
                        modified = true;
                    }
                });
                if (modified) {
                    fs.writeFileSync(fullPath, content, 'utf8');
                    log += `Fixed: ${fullPath}\n`;
                }
            }
        }
    }

    walk(srcPath);
    fs.writeFileSync('repair_log.txt', log + 'Repair Finished');
} catch (e) {
    fs.writeFileSync('repair_error.txt', e.stack);
}

File: repair_encoding.py

import os
import re

# Definindo padrões com escapes unicode para evitar problemas de codificação no próprio script
patterns = [
    # Mojibake (C3 XX)
    (re.compile(r'\u00C3\u00A9'), 'é'),
    (re.compile(r'\u00C3\u00A1'), 'á'),
    (re.compile(r'\u00C3\u00B3'), 'ó'),
    (re.compile(r'\u00C3\u00AA'), 'ê'),
    (re.compile(r'\u00C3\u00A7'), 'ç'),
    (re.compile(r'\u00C3\u00A3'), 'ã'),
    (re.compile(r'\u00C3\u00A0'), 'à'),
    (re.compile(r'\u00C3\u00B5'), 'õ'),
    (re.compile(r'\u00C3\u00BA'), 'ú'),
    (re.compile(r'\u00C3\u00AD'), 'í'),
    (re.compile(r'\u00C3\u0089'), 'É'),
    (re.compile(r'\u00C3\u0081'), 'Á'),
    (re.compile(r'\u00C3\u0093'), 'Ó'),
    (re.compile(r'\u00C3\u0087'), 'Ç'),
    (re.compile(r'\u00C3\u0095'), 'Õ'),
    (re.compile(r'\u00C3\u009A'), 'Ú'),
    (re.compile(r'\u00C3\u008D'), 'Í'),
    
    # Arrows and Emojis
    (re.compile(r'\u00E2\u0086\u0092'), '→'),
    (re.compile(r'\u00E2\u0086\u0091'), '↑'),
    (re.compile(r'\u00E2\u0086\u0093'), '↓'),
    (re.compile(r'\u00E2\u0086\u0090'), '←'),
    (re.compile(r'\u00E2\u009A\u00A0'), '⚠️'),
    
    # Specific Mangles
    (re.compile(r'\u00C0--'), 'x'),
    (re.compile(r'Observaces'), 'Observações'),
    (re.compile(r'movimentaces'), 'movimentações'),
    (re.compile(r'configuraces'), 'configurações'),
    (re.compile(r'operaces'), 'operações'),
    (re.compile(r'notificaces'), 'notificações'),
    (re.compile(r'transacções'), 'transações'),
    (re.compile(r'Transacções'), 'Transações'),
    (re.compile(r'manifestá'), 'manifesto'),
    (re.compile(r'inventrio'), 'inventário'),
    (re.compile(r'prescriptionNão'), 'prescriptionNumber'),
    
    # Missing leading characters
    (re.compile(r"(?<=[ '\"'>])inheiro\b"), 'Dinheiro'),
    (re.compile(r"(?<=[ '\"'>])artão\b"), 'Cartão'),
    (re.compile(r"(?<=[ '\"'>])ransferência\b"), 'Transferência')
]

src_dir = os.path.join(os.getcwd(), 'src')

print(f"Iniciando reparo em: {src_dir}")

fixed_count = 0
for root, dirs, files in os.walk(src_dir):
    if 'node_modules' in dirs: dirs.remove('node_modules')
    if 'dist' in dirs: dirs.remove('dist')
    
    for file in files:
        if file.endswith(('.tsx', '.ts', '.css')):
            full_path = os.path.join(root, file)
            try:
                with open(full_path, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                modified = False
                for pattern, replacement in patterns:
                    if pattern.search(content):
                        content = pattern.sub(replacement, content)
                        modified = True
                
                if modified:
                    with open(full_path, 'w', encoding='utf-8') as f:
                        f.write(content)
                    print(f"Sucesso: {full_path}")
                    fixed_count += 1
            except Exception as e:
                print(f"Erro ao processar {full_path}: {e}")

print(f"Reparo concluído. total de arquivos corrigidos: {fixed_count}")
